### ISSUE TITLE
Querying for a school by name and returning location.lon does not work

### ISSUE THREAD

**Issue by hollyallen** [18F/open-data-maker#227]
When querying for a school by name and wanting to get the location.lon value, the result in an error. Other return values, such as school id, work as expected.

Based on this user report: http://opendata.stackexchange.com/questions/6169/unable-to-get-lat-long-in-api-call

> I am trying to get the latitude and longitude for different schools, and keep getting an error. Location is at the root, and I am able to get ids which are also at the root, so I'm not sure what I'm doing wrong.
> 
> This does not work: https://api.data.gov/ed/collegescorecard/v1/schools?_fields=location.lon&school.name=pepperdine&api_key=**********
> 
> but this does: https://api.data.gov/ed/collegescorecard/v1/schools?_fields=id&school.name=pepperdine&api_key=***********

**Comment by yozlet** [comment]
Apparently we [explicitly exclude the location fields from the `field_types()` method](https://github.com/18F/open-data-maker/blob/dev/lib/data_magic/config.rb#L244) which is why that error's coming up. (And we also [explicitly add a `geo_point` type for `location` whether or not such a field is in the dataset](https://github.com/18F/open-data-maker/blob/dev/lib/data_magic.rb#L139). )

@ultrasaurus: Any idea why we explicitly exclude those fields?

**Comment by slhaines** [comment]
Is it because their searches are not yet adapted to the dot notation: location.lat and location.lon?

**Comment by yozlet** [comment]
@slhaines: No, they're querying the right way, but we're effectively not allowing that for some reason. Also, just using `_fields=location` causes some weird errors in the tests. Will investigate further.

**Comment by ultrasaurus** [comment]
location was implemented as a special case -- the name is hard-coded to "location" and geocoding is handled special for this one field.

**Comment by yozlet** [comment]
Since it may take us a while to get around to fixing, have documented in #283.
